packages/dapp/components/redeem/ucr-nft-redeem.tsx

```diff
@@ -67,12 +67,14 @@ const UcrNftRedeem = () => {
   };
 
   const redeemUcrNftForUad = async (amount: BigNumber) => {
-    const { debtCouponManager } = deployedContracts;
+    const contracts = await protocolContracts;
     const debtId = debtIds[selectedDebtId];
-    if (debtId && (await ensureERC1155Allowance("uCR-NFT -> DebtCouponManager", managedContracts.creditNft, signer, debtCouponManager.address))) {
-      await (await debtCouponManager.connect(signer).redeemCreditNft(debtId, amount)).wait();
-      refreshBalances();
-      fetchDebts(walletAddress, managedContracts.creditNft);
+    if (contracts.creditNft && contracts.creditNftManagerFacet) {
+      if (debtId && (await ensureERC1155Allowance("uCR-NFT -> CreditNftManagerFacet", contracts.creditNft, signer, contracts.creditNftManagerFacet.address))) {
+        await (await contracts.creditNftManagerFacet.connect(signer).redeemCreditNft(debtId, amount)).wait();
```

Comment:
twice `await` why?

Reply:
> twice `await` why?

I'm not sure of it, it was original codebase that I haven't touched

Reply:
> > twice `await` why?
> 
> I'm not sure of it, it was original codebase that I haven't touched

I have just checked, but I can't test this function because the `credit` page is not working due to uniswap router issue. How about check this more after the uniswap credit issue fixed?